homeassistant/components/sense/config_flow.py

```diff
@@ -21,45 +25,118 @@
 )
 
 
-async def validate_input(hass: core.HomeAssistant, data):
-    """Validate the user input allows us to connect.
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Sense."""
 
-    Data has the keys from DATA_SCHEMA with values provided by the user.
-    """
-    timeout = data[CONF_TIMEOUT]
-    client_session = async_get_clientsession(hass)
+    VERSION = 1
 
-    gateway = ASyncSenseable(
-        api_timeout=timeout, wss_timeout=timeout, client_session=client_session
-    )
-    gateway.rate_limit = ACTIVE_UPDATE_RATE
-    await gateway.authenticate(data[CONF_EMAIL], data[CONF_PASSWORD])
+    def __init__(self):
+        """Init Config ."""
+        self._gateway = None
+        self._auth_data = {}
+        super().__init__()
 
-    # Return info that you want to store in the config entry.
-    return {"title": data[CONF_EMAIL]}
+    async def validate_input(self, data):
+        """Validate the user input allows us to connect.
 
+        Data has the keys from DATA_SCHEMA with values provided by the user.
+        """
+        self._auth_data.update(dict(data))
+        timeout = self._auth_data[CONF_TIMEOUT]
```

Comment:
Unrelated to this PR, but we shouldn't ask for a timeout in the config flow.  It should be moved to the options flow, or as developers we should choose a sensible option that works.